src/Compilers/VisualBasic/Portable/CodeGen/EmitConversion.vb

```diff
@@ -230,6 +241,32 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeGen
 
         End Sub
 
+        ''' <summary> 
+        ''' Returns parameterless value type constructor.
+        ''' </summary>
+        Private Function GetParameterlessValueTypeConstructor(typeTo As NamedTypeSymbol) As MethodSymbol
+            Debug.Assert(typeTo.IsValueType AndAlso Not typeTo.IsTypeParameter)
+
+            '  find valuetype parameterless constructor and check the accessibility
+            For Each constr In typeTo.InstanceConstructors
+                ' NOTE: we intentionally skip constructors with all 
+                '       optional parameters; this matches Dev10 behavior
+                If constr.ParameterCount = 0 Then
+                    '  check 'constr' 
+                    If AccessCheck.IsSymbolAccessible(constr, _method.ContainingType, typeTo, useSiteDiagnostics:=Nothing) Then
+                        Return constr
+                    End If
+
+                    '  exit for each in any case
+                    Return Nothing
+                End If
+            Next
+
+            ' This point should not be reachable, because if there is no constructor in the 
+            ' loaded value type, we should have generated a synthesized constructor.
+            Throw ExceptionUtilities.Unreachable
+        End Function
```

Comment:
Consider making this a shared extension method used here and in `LocalRewriter.InitWithParameterlessValueTypeConstructor`.